### PR TITLE
Add custom video/audio transforms

### PR DESCRIPTION
**Pull request by lherman-cs**
I had to update the `MediaStreamConstraints` so that the users will be able to use the APIs nicely.

Resolves https://github.com/pion/mediadevices/issues/31.

### CODE REVIEW

File: examples/transform/main.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	vb.Enable().WithCodec(prop.Codec{
+		CodecName: videoCodecName,
+		BitRate:   100000, // 100kbps
+	})
+	vb.WithProp(prop.Video{
+		FrameFormat: frame.FormatI420, // most of the encoder accepts I420
+		Width:       640,
+		Height:      480,
+	})
+	vb.WithTransforms(removeBlue)
```
Comment (at-wat) [inline examples/transform/main.go]:
```go
	var vb mediadevices.VideoConstraintsBuilder
	vb.Enable().
		WithCodec(prop.Codec{
			CodecName: videoCodecName,
			BitRate:   100000, // 100kbps
		}).
		WithProp(prop.Video{
			FrameFormat: frame.FormatI420, // most of the encoder accepts I420
			Width:       640,
			Height:      480,
		}).
		WithTransforms(removeBlue)
```
would be bit less redundant?

**Comment by lherman-cs** [comment]
> Personally, I like functional options pattern rather than method chaining.
> But using it looks bit complicated due to the namespace.
> Any thoughts?

Yea.. I definitely agree. I don't honestly like method chaining. But, the problem with functional options though is that the users can specify non-related constraints. For example:

```
Video: func(c *MediaTrackConstraints) {
    c.ChannelCount = 2
}
```

I guess the pros and cons would be:

Functional Options:
Pros:
* Definitely look cleaner on the user side
* Less code on our side

Cons:
* User might be confused with the options that they can set because non-related constraints are also included.
* 

Method Chaining or Builder Pattern:
Pros and Cons are basically just the flip side of Functional Options' Pros and Cons.

So, I guess, we'll keep the `MediaTrackConstraints` to be:

```
type MediaTrackConstraints struct {
	prop.Media
	Enabled        bool
	VideoTransform video.TransformFunc
	AudioTransform audio.TransformFunc
}
```

And, we'll keep functional options. What do you think? Do you like the naming and where `VideoTransform` and `AudioTransform` by the way? I'm curious, because I don't really like it, but I don't have a better idea atm.

File: examples/transform/main.go (hunk 1 above)
Comment (lherman-cs) [reply examples/transform/main.go]: Ah, yep. That makes sense.

**Comment by at-wat** [comment]
> So, I guess, we'll keep the `MediaTrackConstraints` to be:
> 
> ```
> type MediaTrackConstraints struct {
> 	prop.Media
> 	Enabled        bool
> 	VideoTransform video.TransformFunc
> 	AudioTransform audio.TransformFunc
> }
> ```
> 
> And, we'll keep functional options. What do you think?

Sounds reasonable to me.

> Do you like the naming and where `VideoTransform` and `AudioTransform` by the way?

They are not actually parts of `Constraints`, but I also don't have good alternative.
So, it's fine for now.

**Comment by lherman-cs** [comment]
> > Do you like the naming and where `VideoTransform` and `AudioTransform` by the way?
> 
> They are not actually parts of `Constraints`, but I also don't have good alternative.
> So, it's fine for now.

Yea.. Definitely agree. It looks weird to be a part of `Constraints`

**Comment by lherman-cs** [comment]
Just updated the PR.

**Pull request merged**